### PR TITLE
Remove support for StatusOr<void>

### LINKED ISSUE #1886: Change functions returning StatusOr<void> to return Status.
The functions in `google::cloud::storage::Client` that return `StatusOr<void>` should return simply `Status`. I will create a separate bug to remove `StatusOr<void>`, consider fixing both at the same time?

### PR DESCRIPTION
**Pull request by devbww**
Change functions that return StatusOr<void> to return just a Status instead, and then remove the StatusOr<void> specialization altogether.  The future of StatusOr<void> is cloudy, so let's avoid it for now.

Fixes #1886 and #1887.

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/1916)

### CODE REVIEW

**Comment by coryan** [comment]
Argh, I should have warned you, there is one remaining usage in `ci/test-install`, unfortunately that directory is only built by one of the CI builds, and not locally (long story, maybe we should figure out a way to fix it).

File: ci/test-install/storage_install_test.cc [hunk 1]
```diff
@@ -47,10 +47,11 @@ int main(int argc, char* argv[]) try {
   while (std::getline(stream, line, '\n')) {
     ++count;
   }
-  client
-      .DeleteObject(bucket_name, object_name,
-                    gcs::Generation(meta.generation()))
-      .value();
+  google::cloud::Status status = client.DeleteObject(
+      bucket_name, object_name, gcs::Generation(meta.generation()));
+  if (!status.ok()) {
```
Comment (remyabel) [inline ci/test-install/storage_install_test.cc]: Not necessarily in the scope of this PR, but I see you adding `.ok()` when I believe we're pushing towards `if (!status)`. Would it save future effort to remove all `.ok()`'s?

Comment (remyabel) [reply ci/test-install/storage_install_test.cc]: Unless `Status` can't do that then ignore this comment.

Comment (devbww) [reply ci/test-install/storage_install_test.cc]: Right ... `Status` can't do that, and I believe the plan of record is that it never will.

**Pull request merged**